metals/src/main/scala/scala/meta/internal/metals/WorkspaceLspService.scala

```diff
@@ -848,10 +848,20 @@ class WorkspaceLspService(
                   }
               }(_ => true)
               .flatMap { mains =>
-                mains.headOption.fold(
+                mains.fold(
```

Comment:
headOption did nothing and was slightly misleading. mains is already an option. There might have been confusion about the return type of Future.find which is Option not a List.